bigtable-hbase-1.x-parent/bigtable-hbase-1.x-mapreduce/src/main/java/com/google/cloud/bigtable/mapreduce/validation/BigtableSyncTableJob.java

```diff
@@ -0,0 +1,483 @@
+          continue;
+        }
+
+        final String targetBigtableAppProfileKey = "--targetbigtableprofile=";
+        if (cmd.startsWith(targetBigtableAppProfileKey)) {
+          targetBigtableAppProfile = cmd.substring(targetBigtableAppProfileKey.length());
+          continue;
+        }
+
+        final String dryRunKey = "--dryrun=";
+        if (cmd.startsWith(dryRunKey)) {
+          BigtableSyncTableAccessor.setDryRun(
+              syncTable, Boolean.parseBoolean(cmd.substring(dryRunKey.length())));
+          continue;
+        }
+
+        final String doDeletesKey = "--doDeletes=";
+        if (cmd.startsWith(doDeletesKey)) {
+          BigtableSyncTableAccessor.setDoDeletes(
+              syncTable, Boolean.parseBoolean(cmd.substring(doDeletesKey.length())));
+          continue;
+        }
+
+        final String doPutsKey = "--doPuts=";
+        if (cmd.startsWith(doPutsKey)) {
+          BigtableSyncTableAccessor.setDoPuts(
+              syncTable, Boolean.parseBoolean(cmd.substring(doPutsKey.length())));
+          continue;
+        }
+
+        final String ignoreTimestampsKey = "--ignoreTimestamps=";
+        if (cmd.startsWith(ignoreTimestampsKey)) {
+          BigtableSyncTableAccessor.setIgnoreTimestamps(
+              syncTable, Boolean.parseBoolean(cmd.substring(ignoreTimestampsKey.length())));
+          continue;
+        }
+
+        printUsage("Invalid argument '" + cmd + "'", args);
+        return false;
+      }
+
+    } catch (Exception e) {
+      e.printStackTrace();
+      printUsage("Can't start because " + e.getMessage(), args);
+      return false;
+    }
+    return true;
+  }
+
+  /**
+   * verify required options and args are correctly set
+   *
+   * @param args
+   * @param sourceBigtableProjectId
+   * @param sourceBigtableInstance
+   * @param targetBigtableProjectId
+   * @param targetBigtableInstance
+   * @return boolean state if required args are set
+   */
+  public boolean verifyRequiredArgsSet(
+      SyncTable syncTable,
+      String[] args,
+      String sourceBigtableProjectId,
+      String sourceBigtableInstance,
+      String targetBigtableProjectId,
+      String targetBigtableInstance) {
+    if (BigtableSyncTableAccessor.getSourceZkCluster(syncTable) == null
+        && (sourceBigtableProjectId == null || sourceBigtableInstance == null)) {
+      printUsage(
+          "--sourcezkcluster or --sourcebigtableproject and --sourcebigtableinstance required.",
+          args);
+      return false;
+    }
+
+    if (BigtableSyncTableAccessor.getTargetZkCluster(syncTable) == null
+        && (targetBigtableProjectId == null || targetBigtableInstance == null)) {
+      printUsage(
+          "--targetzkcluster or -targetbigtableproject and --targetbigtableinstance required.",
+          args);
+      return false;
+    }
+
+    if (BigtableSyncTableAccessor.getSourceHashDir(syncTable) == null) {
+      printUsage("--sourcehashdir is required.", args);
```

Comment:
I don't think the input we require has the --sourceHashDir prefix. The 3 required params are not prefixed by their names. We should remove this "language" in the log and add unit tests where these are not passed

Reply:
great catch, this inherits behavior from HBase so we haven't changed the input args vs. prefixed `--` config params.